port/unix/omrsignal.c

```diff
@@ -596,6 +605,24 @@ omrsig_register_os_handler(struct OMRPortLibrary *portLibrary, uint32_t portlibS
 	} else {
 		omrthread_monitor_enter(registerHandlerMonitor);
 		rc = registerSignalHandlerWithOS(portLibrary, portlibSignalFlag, (unix_sigaction)newOSHandler, oldOSHandler);
+		if (0 == rc) {
+			/*  A user-specified handler has been successfully registered for a signal. */
+			if ((newOSHandler == (void *)masterSynchSignalHandler)
+				|| (newOSHandler == (void *)masterASynchSignalHandler)
+			) {
+				/* User-specified handler is a master handler. So, we should set the portlibSignalFlag bit in
+				 * signalsWithMasterHandlers.
+				 */
+				signalsWithMasterHandlers |= portlibSignalFlag;
+			} else {
+				/* If the user-specified handler is not a master handler, then we should un-set the
```

Comment:
avoid "we" and "will". There are more places, I'm not going to point them all out.